benchmark-service/benchmark/benchmark.py

```diff
@@ -374,7 +375,7 @@ async def update_commit(app, sha):  # pylint: disable=unused-argument
         return commit
 
     has_results_file = gs_reader.file_exists(file_path)
-    if has_results_file and commit['date'] not in benchmark_data['dates']:
+    if has_results_file and benchmark_data['commits'].get(sha) is not None:
```

Comment:
I think this is clearer as `if has_results_file and sha in benchmark_data['commits']`